File: rzops-web/src/lib/types/change_record.ts
export interface ChangeRecordResponse {
  id: string;
  actor_id: string | null;
  change_type: string;
  resource_type: string | null;
  resource_id: string | null;
  actor_email: string | null;
  before_data: unknown;
  after_data: unknown;
  remarks: string | null;
  created_at: string;
}

export interface ChangeRecordListResponse {
  data: ChangeRecordResponse[];
  count: number;
}

export interface ListChangeRecordsQuery {
  change_type?: string;
  resource_type?: string;
  actor_id?: string;
  created_from?: string;
  created_to?: string;
  page?: number;
  per_page?: number;
}

File: rzops-api/domain/src/models/change_record.rs
use chrono::{DateTime, Utc};
use serde_json::Value;
use uuid::Uuid;

use crate::enums::ChangeType;

/// ChangeRecord entity — maps to cmdb_change_record.
#[derive(Debug, Clone)]
pub struct ChangeRecord {
    pub id: Uuid,
    pub actor_id: Option<Uuid>,
    pub change_type: ChangeType,
    pub resource_type: String,
    pub resource_id: Option<Uuid>,
    pub before_data: Value,
    pub after_data: Value,
    pub remarks: Option<String>,
    pub created_at: DateTime<Utc>,
}
